### PR TITLE
Move build to dotnet

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
+    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetvolatiledev/api/v3/index.json" />
```
Comment (victorhurdugaci) [inline NuGet.config]: don't forget to revert this

File: test/Microsoft.EntityFrameworkCore.FunctionalTests/project.json [hunk 2]
```diff
@@ -41,14 +46,6 @@
       "dependencies": {
         "xunit.runner.aspnet": "2.0.0-aspnet-*"
```
Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.FunctionalTests/project.json]: This should be gone if you're running tests with dotnet test

File: test/Microsoft.EntityFrameworkCore.InMemory.Tests/project.json [hunk 3]
```diff
@@ -7,16 +7,14 @@
...
   },
   "testRunner": "xunit",
-  "commands": {
-    "test": "xunit.runner.aspnet"
-  },
   "frameworks": {
-    "dnx451": { },
     "dnxcore50": {
-      "imports": "portable-net452+win81",
+      "imports": "portable-net451+win8",
```
Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.InMemory.Tests/project.json]: @bricelam is this change okay?

**Comment by bricelam** [comment]
This will break running unit tests form Visual Studio on the RC1 tooling. cc @divega

File: test/Microsoft.EntityFrameworkCore.Microbenchmarks.Core/project.json [hunk 4]
```diff
@@ -6,16 +6,13 @@
...
-  },
   "frameworks": {
     "dnxcore50": {
-      "imports": "portable-net452+win81",
+      "imports": "portable-net451+win8",
       "dependencies": {
         "Microsoft.NETCore.Platforms": "1.0.1-*",
         "System.Data.SqlClient": "4.0.0-*",
-        "xunit.runner.aspnet": "2.0.0-aspnet-*"
+        "dotnet-test-xunit": "1.0.0-dev-*"
```
Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.Microbenchmarks.Core/project.json]: Line 20 in this file shouldn't be there anymore

File: test/Microsoft.EntityFrameworkCore.InMemory.Tests/project.json (hunk 3 above)
Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.InMemory.Tests/project.json]: Seems unnecessary, but fine.

File: test/Microsoft.EntityFrameworkCore.Microbenchmarks.EF6/project.json [hunk 5]
```diff
@@ -5,9 +5,6 @@
   },
   "content": "config.json",
   "testRunner": "xunit",
-  "commands": {
-    "test": "xunit.runner.aspnet"
-  },
   "frameworks": {
     "dnx451": {
       "dependencies": {
```
Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.Microbenchmarks.EF6/project.json]: xunit.runner.aspnet should be gone from dependecies

File: test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json [hunk 6]
```diff
@@ -11,6 +11,14 @@
     "Microsoft.Dnx.Compilation.CSharp.Abstractions": "1.0.0-*"
   },
   "frameworks": {
+    "dnxcore50": {
+      "imports": "portable-net451+win8",
+      "dependencies": {
+        "Microsoft.Dnx.Runtime": "1.0.0-*",
```
Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json]: Ugh... Probably not related but this dependency should go away. @bricelam ?

Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json]: Same for line 44

File: test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json [hunk 7]
```diff
@@ -11,6 +11,14 @@
     "Microsoft.Dnx.Compilation.CSharp.Abstractions": "1.0.0-*"
```
Comment (victorhurdugaci) [inline test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json]: This shouldn't be here

**Comment by victorhurdugaci** [comment]
:watch:

File: test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json (hunk 6 above)
Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json]: Not sure what it's used for. @natemcmaster or @lajones might know.

Comment (lajones) [inline test/Microsoft.EntityFrameworkCore.Relational.Design.FunctionalTests/project.json]: @bricelam @victorhurdugaci  I'm speculating, but after we have scaffolded the .cs files we go ahead and compile using Roslyn just to check that the C# we're producing compiles. Could this reference be for that?

File: test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/QuerySqlServerTest.cs [hunk 8]
```diff
@@ -2211,6 +2211,7 @@ public override void Join_composite_key()
                 Sql);
         }
 
+        [Fact(Skip = "Fix MetadataReferencesProvider.AddReferenceFromName for dnxcore50")]
```
Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/QuerySqlServerTest.cs]: Wait. These tests don't use `MetadataReferencesProvider`...

**Comment by natemcmaster** [comment]
Superceded by #4638

**Pull request closed without merging**